test/gtest/uct/ib/test_ib.cc

```diff
@@ -102,6 +102,35 @@ class test_uct_ib_addr : public test_uct_ib {
 
         free(ib_addr);
     }
+
+    void test_fill_ah_attr(uint64_t subnet_prefix) {
+        uct_ib_iface_t *iface     = ucs_derived_of(m_e1->iface(), uct_ib_iface_t);
+        static const uint16_t lid = 0x1ee7;
+        union ibv_gid gid;
+        struct ibv_ah_attr ah_attr;
+
+        gid.global.subnet_prefix = subnet_prefix ?: iface->gid.global.subnet_prefix;
+        gid.global.interface_id  = 0xdeadbeef;
+
+        if ((iface->gid.global.subnet_prefix != UCT_IB_LINK_LOCAL_PREFIX) &&
+            (gid.global.subnet_prefix == UCT_IB_LINK_LOCAL_PREFIX)) {
+            UCS_TEST_SKIP_R("not applicable");
+        }
+
+        uct_ib_iface_fill_ah_attr_from_gid_lid(iface, lid, &gid, &ah_attr);
+
+        if (uct_ib_iface_is_roce(iface)) {
+            EXPECT_TRUE(ah_attr.is_global == 1);
```

Comment:
1. we can use EXPECT_EQ
2. we need some case where we EXPECT_EQ(0, ah_attr.is_global) to make sure grh is NOT used

Reply:
EXPECT_EQ is used.
check for is_global is not used is implemented in line 128:
```
} else if (iface->gid.global.subnet_prefix == gid.global.subnet_prefix) {
            EXPECT_EQ(ah_attr.is_global, iface->is_global_addr);
```

Reply:
what if  iface->is_global_addr is buggy?
we need to compare to 0

Reply:
do you mean iface->is_global_addr>1?
ok, fixed

Reply:
i mean, the test should be more robust, it should not rely on iface logic, instead it should compare iface logic result to some expected value.

Reply:
result of logic is ah_attr which destroyed after connection complete... how can we check it?
this test checks for ah_attr filled correctly